api/src/Responder/Responder/ApiResponder.php

```diff
@@ -137,16 +142,37 @@ protected function collectionWithStatusCode($items, $code = 200)
     }
 
     /**
-     * TODO: Implement paginator() method.
-     * Bind a paginator to a transformer and start building a response.
+     * Build paginated response.
      *
-     * @param Paginator $paginator
+     * @param Collection|array $items
+     * @param null|int $offset
+     * @param null|int $totalCount
      * @param array $parameters
      *
      * @return Response
      */
-    public function paginator(Paginator $paginator, array $parameters = [])
+    public function paginatedCollection($items, $offset = null, $totalCount = null, array $parameters = [])
     {
+        $response = $this->getResponse();
+        $response->headers->set('Content-Type', 'application/json');
+        $offset = is_null($offset)?0:$offset;
+        $totalCount = is_null($totalCount)?'*':$totalCount;
+        $itemCount = count($items);
+        $rangeHeader = $offset.'-'.($itemCount+$offset-1).'/'.$totalCount;
+        $response->headers->set('Content-Range', $rangeHeader);
+        $response->setContent($this->encode($this->getTransformer()->transformCollection($items)));
+
+        if ($this->request->headers && $this->request->headers->has('Range')) {
+            if ($itemCount > 0) {
+                $response->setStatusCode(206);
+            } else {
+                $response->setStatusCode(416, 'Requested Range Not Satisfiable');
+            }
+        } else {
+            $response->setStatusCode(200);
+        }
+
+        return $response;
     }
 
```

Comment:
The client should be able to send an options request to a collection to get the `Accept-Ranges` header, so it knows what the name of the entity is so it can make the appropriate request